independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanConfiguratorBase.java

```diff
@@ -201,12 +201,22 @@ public THIS targetPackageName(String name) {
         return self();
     }
 
+    /**
+     * @deprecated use {@link #alternative(boolean)} and {@link #priority(int)};
+     *             this method will be removed at some time after Quarkus 3.6
```

Comment:
I would recommend to rather use `since = "3.0"` in the `@Deprecated` annotation.

We never really give estimate of when something will be removed. But us knowing when it was removed is useful for knowing when it's relatively safe to remove something.

Reply:
We actually often do that in ArC, but adding `since` is a great point, I'll do that.

Reply:
+1 for `since`

Reply:
@Deprecated(forRemoval = true, since = "3.0")